### PR TITLE
Refactor/api calls

### LINKED ISSUE #31: Refactor saveFavorite to incorporate POST

### LINKED ISSUE #32: Refactor Favorites to fetch from posters endpoint instead of mockData

### LINKED ISSUE #33: GET quotes, images, and posters on page load

### LINKED ISSUE #35: As a user, when I add a poster to favorites I see a message confirming it has been added

### LINKED ISSUE #41: Error handling
- [x] UI error handling for fetch calls
- [x] Sad path message for no favorites
- [x] #45
- [x] #44
- [x] Saving duplicates
- [x] Sad path for bad URL

### LINKED ISSUE #42: Background color is not taking up the full height of the favorites page

### PR DESCRIPTION
**Pull request by rachelsoae**
What I did:
- Added GET and POST requests
- Replaced mock data with live data
- Added toggle feature so that when a user clicks to save a favorite, state is updated and the button changes to a non-clickable element saying 'Saved!'
- Created an error element with Navigation to error route/element
- Fixed a couple of small styling bugs
- Added a message for empty favorites array - DELETED all favorites in Postman

resolves #31 
resolves #32 
resolves #33 
resolves #35 
resolves #42 
resolves #41  - unresolved bulletpoints have been converted to new issue (see below)

Did this break anything?
- [x] No
- [ ] Yes

Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ] Super small fix (Corrected a typo, removed a comment, etc.)
- [ ] Skip all the other stuff and briefly explain the fix.

Checklist:
- [ ] If this code needs to be tested, all tests are passing
- [x] The code runs locally
- [x] There are comments where clarification/ organization is needed
- [x] The code is DRY. If it's not, I tried my best
- [x] I reviewed my code before pushing

What's next?
I need help coming up with error handling/sad paths for the form. I will turn this into a sub-issue of error handling:
- Currently, if the poster type is not selected, clicking 'create' will route to the error page, because the link to the poster page is dependent on that selection
- As long as one of those buttons is clicked, the user can still submit the form with blank input and get a blank poster
- If they put random characters (not a valid URL) in the image field, the quote says "undefined"
Because of the way the Link is structured on the 'create' button, I'm not sure of the best solution. Surely some kind of state or conditional will resolve it - but it's late, so we can try again tomorrow.

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -34,60 +48,42 @@ const App = () => {
...
+        <Route 
+          path='/poster/:type' 
+          element={<Poster 
+            poster={poster} 
+            font={fontSize} 
+            saveFavorite={saveFavorite} 
+            randomizePoster={randomizePoster}  
+            isFavorite={isFavorite} 
+            />}/>
+        <Route path='*' element={<Error error={error} />} />
```
Comment (dcochico) [inline src/App/App.js:86]: Nice 404 error handling!

File: src/App/App.js [hunk 2]
```diff
@@ -1,29 +1,43 @@
...
+    getData('quotes')
+    .then(response => setQuotes(response.quotes))
+    .catch(error => setError(error.message))
 
-  //   getData(images)
-  //   .then(response => setImages(response.images))
-  //   .catch(error => alert(error.message))
-  // }, [])
+    getData('images')
+    .then(response => setImages(response))
```
Comment (dcochico) [inline src/App/App.js:27]: interesting that this still works. I would imagine this line would have to be `.then(response => setImages(response.images))`

File: src/App/App.js [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
+    .catch(error => setError(error.message))
 
+    getData('posters')
+    .then(response => setFavorites(response))
```
Comment (dcochico) [inline src/App/App.js:31]: same with this, interesting that this didn't need to be `.then(response => setImages(response.posters))`

File: src/Error/Error.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import './Error.css'
+import '../Nav/Nav.css'
+import {NavLink} from 'react-router-dom'
+
+const Error = ({error}) => {
+  return (
+    <main className='main-error'>
+      <div className='error-container'>
+        <h2 className='error-message'>🤕  Uh-oh... There's been an error  🤕</h2>
```
Comment (dcochico) [inline src/Error/Error.js:9]: well done

File: src/Poster/Poster.js [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
     <main className={`poster poster-${poster.type}`}>
       <div className='poster-container'>
-        <Card image={poster.image} quote={poster.quote} type={poster.type} font={font} />
+        <Card image={poster.src} quote={poster.quote} type={poster.type} font={font} />
       </div> 
       <div className='poster-buttons'>
-        <button className='poster-button' onClick={saveFavorite}>💛 save to favorites</button>
         <button className='poster-button' onClick={() => randomizePoster('wholesome')}>🥹 random wholesome</button>
         <button className='poster-button' onClick={() => randomizePoster('chaotic')}>😈 random chaotic</button>
+        {isFavorite ? <p className='poster-button saved'>Saved!</p> : <button className='poster-button' onClick={saveFavorite}>💛 save to favorites</button>}
```
Comment (dcochico) [inline src/Poster/Poster.js:14]: great use of conditional rendering

File: src/apiCalls.js
Comment (dcochico) [inline src/apiCalls.js:1]: api calls work!! success

**Review by dcochico — APPROVED** [review]
reviewed and left comments

**Review by JamieCaudill — COMMENTED** [review]
Pulled the branch down and ran the code. It looks good! Nice error handling!

**Pull request merged**